src/Magnum/EigenIntegration/Integration.h

```diff
@@ -369,6 +370,51 @@ template<class To, std::size_t size, class T> inline To cast(const Math::Vector<
     return Math::Implementation::VectorConverter<size, T, To>::to(from);
 }
 
+
+/**
+@brief Convert Corrades StridedArrayView2D to Eigens Dynamic Matrix Type.
+*/
+template<class T> inline Eigen::Map<Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>, Eigen::Unaligned, Eigen::Stride<Eigen::Dynamic, Eigen::Dynamic>> arrayCast(const Containers::StridedArrayView2D<T>& from) {
+    using MatrixT = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+    using StrideT = Eigen::Stride<Eigen::Dynamic, Eigen::Dynamic>;
+    auto size = from.size();
+    auto stride = from.stride();
+    return Eigen::Map<MatrixT, Eigen::Unaligned, StrideT>(reinterpret_cast<T*>(from.data()), size[0], size[1], StrideT(stride[0]/sizeof(T), stride[1]/sizeof(T)));
+}
+
+/**
+@brief Convert Corrades StridedArrayView1D to Eigens Dynamic Vector Type.
+Since for a one dimensional StridedArrayView there is no column or row version
+we always return an Eigen Column Vector.
+*/
+template<class T> inline Eigen::Map<Eigen::Matrix<T, Eigen::Dynamic, 1>, Eigen::Unaligned, Eigen::InnerStride<>> arrayCast(const Containers::StridedArrayView1D<T>& from) {
+    using MatrixT = Eigen::Matrix<T, Eigen::Dynamic, 1>;
+    return Eigen::Map<MatrixT, Eigen::Unaligned, Eigen::InnerStride<>>(reinterpret_cast<T*>(from.data()), from.size(), Eigen::InnerStride<>(from.stride()/sizeof(T)));
+}
+
+/**
+@brief Convert an Eigen Expression to Corrades StridedArrayView1D
+If it is known at compile time that the Eigen Expression has either
+one Column or one Row, then this function maps the Eigen Expression
+to a one dimensional StridedArrayView.
+*/
+template<class Derived> inline typename std::enable_if<(Eigen::internal::traits<Derived>::ColsAtCompileTime == 1) || (Eigen::internal::traits<Derived>::RowsAtCompileTime == 1), Containers::StridedArrayView1D<typename Derived::Scalar>>::type arrayCast(const Eigen::DenseCoeffsBase<Derived, Eigen::DirectWriteAccessors>& from) {
+    using Scalar = typename Derived::Scalar;
+    return Containers::StridedArrayView1D<Scalar>({const_cast<Scalar*>(&from(0,0)), std::size_t(from.size()*from.innerStride())}, std::size_t(from.size()), std::size_t(from.innerStride())*sizeof(Scalar));
+}
+
+/**
+@brief Convert an Eigen Expression to Corrades StridedArrayView2D
+Overload that takes care of any Eigen Expression that was not handled by @ref Magnum::EigenIntegration::arrayCast().
+@snippet EigenIntegration.cpp IntegrationStridedArrayView
+*/
+template<class Derived> inline typename std::enable_if<(Eigen::internal::traits<Derived>::ColsAtCompileTime == Eigen::Dynamic) && (Eigen::internal::traits<Derived>::RowsAtCompileTime == Eigen::Dynamic), Containers::StridedArrayView2D<typename Derived::Scalar>>::type arrayCast(const Eigen::DenseCoeffsBase<Derived, Eigen::DirectWriteAccessors>& from) {
+    using Scalar = typename Derived::Scalar;
+    /* we assume that the Eigen expression is a slice of some continuous piece of memory */
```

Comment:
Err... not happy about this comment :sweat_smile: 

So if I do a `image.pixels<Color4>().flipped<0>()` (i.e., image pixels with Y flipped), pass it to Eigen to do whatever magic something, and then convert back to a `StridedArrayView2D`, what do I get? This is one of the things that's best to either have implement fully (so there are no surprises when one gets slightly advanced) or not at all.

Looking further, the `size` is being used only in the first argument, where Corrade needs it only for an assert. I think we could safely assume the Eigen memory has correct bounds, and sidestep the assert by passing `~std::size_t{}` instead of the `size`.